patches/server/1046-Brigadier-based-command-API.patch

```diff
@@ -0,0 +1,2415 @@
++                return sourceStack -> Collections2.transform(result.getNames((CommandSourceStack) sourceStack), CraftPlayerProfile::new);
++            } else {
++                return sourceStack -> Collections2.transform(result.getNames((CommandSourceStack) sourceStack), CraftPlayerProfile::new);
++            }
++        });
++    }
++
++    @Override
++    public ArgumentType<BlockPositionResolver> blockPos() {
++        return this.of(BlockPosArgument.blockPos(), (result) -> sourceStack -> {
++            BlockPos pos = result.getBlockPos((CommandSourceStack) sourceStack);
++
++            return Position.block(pos.getX(), pos.getY(), pos.getZ());
++        });
++    }
++
++    @Override
++    public ArgumentType<BlockState> blockState() {
++        return this.of(BlockStateArgument.block(PaperCommands.INSTANCE.getBuildContext()), (result) -> {
++            return CraftBlockStates.getBlockState(result.getState(), result.tag);
++        });
++    }
++
++    @Override
++    public ArgumentType<ItemStack> itemStack() {
++        return this.of(ItemArgument.item(PaperCommands.INSTANCE.getBuildContext()), (result) -> {
++            return CraftItemStack.asBukkitCopy(result.createItemStack(1, true));
++        });
++    }
++
++    @Override
++    public ArgumentType<NamedTextColor> namedColor() {
++        return this.of(ColorArgument.color(), result ->
++            requireNonNull(
++                NamedTextColor.namedColor(
++                    requireNonNull(result.getColor(), () -> result + " didn't have a color")
++                ),
++                () -> result.getColor() + " didn't map to an adventure named color"
++            )
++        );
++    }
++
++    @Override
++    public ArgumentType<Component> component() {
++        return this.of(ComponentArgument.textComponent(), PaperAdventure::asAdventure);
++    }
++
++    @Override
++    public ArgumentType<MessageArgumentResponse> signedMessage() {
++        return this.of(MessageArgument.message(), MessageArgumentImpl::new);
++    }
++
++    @Override
++    public ArgumentType<DisplaySlot> scoreboardDisplaySlot() {
++        return this.of(ScoreboardSlotArgument.displaySlot(), CraftScoreboardTranslations::toBukkitSlot);
++    }
++
++    @Override
++    public ArgumentType<NamespacedKey> namespacedKey() {
++        return this.of(ResourceLocationArgument.id(), CraftNamespacedKey::fromMinecraft);
++    }
++
++    @Override
++    public ArgumentType<Key> key() {
++        return this.of(ResourceLocationArgument.id(), CraftNamespacedKey::fromMinecraft);
++    }
++
++    @Override
++    public ArgumentType<Range<Integer>> integerRange() {
++        return this.of(RangeArgument.intRange(), VanillaArgumentProviderImpl::convertToRange);
++    }
++
++    @Override
++    public ArgumentType<Range<Double>> doubleRange() {
++        return this.of(RangeArgument.floatRange(), VanillaArgumentProviderImpl::convertToRange);
```

Comment:
I cannot find this discussed anywhere. Admittedly, I cannot think of a scenario where the difference matters. Minecraft rarely if ever deals with numbers large enough to matter. But just wondered why there is the discrepancy between float internally and double in the api here.

Again, I don't know when this would matter - so not necessarily suggesting a change.